src/diffusers/models/unet_2d_blocks.py

```diff
@@ -1658,8 +1695,10 @@ def custom_forward(*inputs):
                     create_custom_forward(attn, return_dict=False),
                     hidden_states,
                     encoder_hidden_states,
+                    temb,
                     attention_mask,
                     cross_attention_kwargs,
+                    encoder_attention_mask,
```

Comment:
This addition is reasonable. But seems like we weren't passing `temb` previously here which seems a bit weird to me. @williamberman any comments here?

Reply:
Actually, yes this is wrong thanks for the catch. @Birch-san can you remove passing the time embedding to the attention block here and anywhere else it's added? The standard unets we use condition on time embeddings through the resnet and text embeddings through the attention blocks

Reply:
I can remove the passing of `temb` again, but I added it because @patrickvonplaten asked for it…  
https://github.com/huggingface/diffusers/pull/2634#discussion_r1191318024  
https://github.com/huggingface/diffusers/pull/2634#discussion_r1191318530

Reply:
I've removed the passing of `temb` that I added to the checkpointed attn in those two places (`KCrossAttn{Up,Down}Block2D#forward`), **and** I removed the existing `emb=temb` which was originally being passed to the non-checkpointed attn.

Reply:
@williamberman removing the pre-existing `emb=temb` arg to (non-checkpointed) attn in `KCrossAttn{Up,Down}Block2D#forward`, resulted in test failure:  
https://github.com/huggingface/diffusers/actions/runs/5019356774/jobs/8999754308?pr=2634

since invoking `KAttentionBlock#forward` with `emb=None` causes `self.norm2(hidden_states, emb)` to go bang.

I don't know how to proceed. it seems like the non-checkpointed attn() calls **needed** those `emb=temb` arguments.  
if I put `KCrossAttn{Up,Down}Block2D#forward` back to how main looks (non-checkpointed attn() calls would pass `emb=temb` to `KAttentionBlock#forward`: it raises the question of whether `emb` should be passed into their corresponding _checkpointed_ attn() calls.  
in main, the checkpointed calls are just wrong (positional arguments don't line up correctly with `KAttentionBlock#forward`, and `emb` is not passed, which means it doesn't match the non-checkpointed call).

basically it feels to me like Patrick's suggestion (make the checkpointed calls do the same thing the non-checkpointed calls do) made sense. I'm trying to understand what your suggestion is, since my attempt at interpreting this led to test failures.

Reply:
I've put this code back to how it was when this review thread started. with the original `emb=temb` back in for non-checkpointed calls (on the basis that this is required in order for tests to pass), and my addition (`emb` for checkpointed calls) back in too (on the basis that surely checkpointed calls should match non-checkpointed calls).

Reply:
Think this is good as is

Reply:
Ah sorry! Didn't realize there was the separate k attention block here, thought it was the regular attention block, no worries :)